---
title: "Section 9: LATENT DIRICHLET ALLOCATION"
author: "Pratheepa Jeganathan"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: BiocStyle::html_document
params:
  K: 11
  R: 2000
---

The input parameters to run this R Markdown are $K$ number of topics and R - number of interations for each Markov chain.

We removed DNA contaminants using BARBI. Then, we choose ASVs that have at least 25 counts in at least two specimens.

In microbiome data, bacteria that have similar functionalities can co-occure as latent communities. In this data set, we are interested infer on the latent communties when we use original and modified PNA clams. To infer about these latent communities, we use topic modeling.  

For the topic modeling, we choose Aster-plants. These specimens were sequenced with original and modified pPNA types.

We set the hyper-parameters $\alpha$ and $\gamma$ less than one to generate mixtures that are different from each other. We choose 0.8 for $\alpha$ across all samples so that we avoid generating unrealistic topics. 

We estimate the parameters using HMC NUTS with four chains and 2000 iterations. Out of these 2000 iterations, 1000 iterations were used as warmup samples. 

We align the topics from four different chain and compute posterior log likelihood, $\hat{R}$ (split), effective sample size (ESS). 

For the model with optimal topic, we do the predictive model check with simulated data, observed data, and a statistic $T\left(K_{ij} \right)$

With the optimal topic, we visualize the topic distribution in each specimen and ASVs distribution in each topic.

With the optimal topic, we do differential topic analysis

```{r read_arg}
K <- params$K
K
iter <- params$R
iter
```


```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
  fig.width = 7, 
  fig.height = 4, 
  fig.align = 'center',
  message = FALSE, 
  warning = FALSE)
```

```{r}
library(phyloseq)
library(tidyverse)
library(genefilter) #KOverA
library(Rcpp)
library(rstan)

devtools::load_all()
```


```{r}
theme_set(theme_minimal())
theme_update(
  text = element_text(size = 10),
  legend.text = element_text(size = 10)
)
```

## Data

```{r}

threshold <- kOverA(2, A = 25) 
psE_BARBI <- phyloseq::filter_taxa(
  psE_BARBI, 
  threshold, TRUE) 

psE_BARBI

ps <- psE_BARBI
rm(psE_BARBI)
ps <- prune_taxa(taxa_sums(ps) > 0, ps)
ps

```

### Edit specimen names

We edit specimen names and identify Asteraceae and non-Asteraceae plants.

```{r}
sam_names <- str_replace(sample_names(ps), "E106", "E-106")
sam_names <- str_replace(sam_names, "_F_filt.fastq.gz", "")
sam_names <- str_replace(sam_names, "Connor-", "E")
sample_names(ps) <- sam_names
sample_data(ps)$X <- sam_names
sample_data(ps)$unique_names <- sam_names
aster <- c("142","143","15","ST","22","40")
non_aster <- c("33", "71", "106")

paired_aster <- c("E-142-1", "E142-1", "E-142-5", "E142-5", "E-142-10", "E142-10", "E-143-2", "E143-2", "E-143-7", "E143-7", "E-15-1", "E15-1", "ST-CAZ-4-R-O", "ST-CAZ-4-R-M", "ST-SAL-22-R-O", "ST-SAL-22-R-M", "ST-TRI-10-R-O", "ST-TRI-10-R-M")
paired_non_aster <- c("E33-7", "E-33-7", "E33-8", "E-33-8", "E33-9", "E-33-9", "E71-10", "E-71-10","E71-2","E-71-2" ,"E71-3", "E-71-3", "E106-1", "E-106-1", "E106-3", "E-106-3", "E106-4", "E-106-4")
paired_specimens <- c(paired_aster, paired_non_aster)
```


```{r}
# We will use 9 colors for 9 different plants
plant_colors <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "purple")
```

## Topic modeling

### Set-up data
```{r}
x <- t(get_taxa(ps))
dimnames(x) <- NULL

# theta[d] ~ dirichlet(alpha), alpha pseudocount for each topic
# beta[k] ~ dirichlet(gamma), gamma pseudocount for each ASV in each topic
print(K)
stan.data <- list(K = K, 
  V = ncol(x), 
  D = nrow(x), 
  n = x, 
  alpha = rep(.8, K), 
  gamma = rep(.5, ncol(x))
)
```

### Posterior sampling 
$\theta$ and $B = \left[\beta_{1}, \cdots, \beta_{K} \right]^{T}.$

```{r eval=FALSE}
fileN <- paste0(
  "../results/JABES_Endo_all_K_",
  K,
  "_ite_",
  iter,
  ".RData"
  )

stan.fit <- stan(file = "./lda.stan", 
  data = stan.data, 
  iter = iter, 
  chains = 4, 
  sample_file = NULL,
  diagnostic_file = NULL,
  cores = 4,
  control = list(adapt_delta = 0.9),
  save_dso = TRUE,
  algorithm = "NUTS")

save(stan.fit, file = fileN)
```

```{r eval=FALSE}
load(file = fileN)
```


### Extract posterior samples
```{r eval=FALSE}
samples <- rstan::extract(
  stan.fit, 
  permuted = TRUE, 
  inc_warmup = FALSE, 
  include = TRUE)# samples is a list
```

### Alignment
- Create a Topic $*$ Chain matrix
```{r eval=FALSE}
theta <- samples$theta 
# dim(theta)
aligned <- alignmentMatrix(
  theta, 
  ps, 
  K, 
  iter = iter,
  chain = 4,
  SampleID_name = "unique_names"
  )

theta_aligned <- thetaAligned(
  theta, 
  K, 
  aligned, 
  iter = iter, 
  chain = 4
  )

dimnames(theta_aligned)[[2]] <- sample_names(ps)
dimnames(theta_aligned)[[3]] <- c(paste0("Topic_", seq(1,K)))

# array to a dataframe
theta_all <- reshape2::melt(theta_aligned)

colnames(theta_all) <- c(
"iteration", 
"Sample", 
"Topic", 
"topic.dis"
)

theta_all$Chain <- paste0(
  "Chain ", 
  rep(seq(1, 4), 
      each = (iter/2)
      )
  )

sam <- sample_data(ps) %>% 
  data.frame()

theta_all$Sample <- theta_all$Sample %>%
  as.character()

theta_all <- left_join(
  theta_all, 
  sam, 
  by =c("Sample"= "unique_names")
  )

theta_all$Chain <- factor(theta_all$Chain)
theta_all$Topic <- factor(theta_all$Topic)
theta_all$Sample <- factor(theta_all$Sample)
theta_all$pna <- factor(theta_all$pna)
```

### Plot the topic distribution in all chains

- facet by pna and aster or non-aster plant type.
- In the filtered ps, we have only one aster plant type
```{r eval=FALSE}
theta_summary <- theta_all %>% 
  group_by(
    Sample, 
    Topic, 
    pna
    ) %>% 
  summarize(
    median.topic.dis = median(topic.dis)
    ) %>% 
  ungroup() %>% 
  mutate(
    Topic = factor(
      Topic, 
      levels = rev(str_c("Topic_",1:K)
                   )
      )
    )

p <- ggplot(
  theta_summary, 
  aes(x = pna, 
      y = Topic,
      fill = pna)
  )
p <- p+
  geom_tile(
    aes(alpha = median.topic.dis)
    ) +
  facet_grid(
    .~Sample,
    scale = "free"
    )+
  xlab("pna") +
  scale_fill_manual(
    name = "pna", 
    values = c("steelblue1","green3")
    ) +
  scale_alpha(
    name = "median topic distribution"
    ) +
  theme_minimal(
    base_size = 20
    ) +  
  theme(
    plot.title = element_text(hjust = 0.5),
    strip.text.x = element_text(angle = 90)
    )
p
ggsave(paste0("topic_dis_all_",K, ".png")), p, width = 30, height = 16)
```

```{r echo=FALSE, out.width="70%", fig.caption = "Figure 6", fig.align="center"}
knitr::include_graphics(paste0("topic_dis_all_",K,".png"))
```

```{r include=FALSE}
rm(params, ps, stan.data, x, aster, iter, K, non_aster, paired_aster, paired_non_aster, paired_specimens, plant_colors, sam_names)
```


