---
title: "Analzying molecular microbiome data with diffTop"
author: "Pratheepa Jeganathan and Susan Holmes"
date: "`r format(Sys.Date(), '%m/%d/%Y')`"
abstract: >
  One of the important tasks in molecular microbiome data is the quantification and statistical inference  of micriobial communities across envrionment or groups. The count data are presented as a contingency table, for each sample the number of sequence fragments that have been assigned to each taxon. The package diffTop provides methods to estimate topics using latent Dirichlet allocation, align posterior samples across chains, find topic differences across environments by use of generalized linear models, visualize the topic and taxa proportions. This vignette explains the use of the package and demonstrates typical workflows.
output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    fig_width: 5
bibliography: library.bib
vignette: >
  %\VignetteIndexEntry{Analzying molecular microbiome data with diffTop}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  %\usepackage[utf8]{inputenc}
---

```{r setup, echo=FALSE, results="hide"}
knitr::opts_chunk$set(tidy = FALSE,
                      cache = FALSE,
                      dev = "png",
                      message = FALSE, error = FALSE, warning = TRUE)
```	

```{r}
devtools::load_all()
# library(diffTop)

```

# Standard workflow

**Note:** if you use diffTop in published research, please cite:

> Jeganathan, P. and Holmes, S.P. (2021)
> A statistical perspective on the challenges in molecular microbial biology.
> *Journal of Agricultural, Biological and Environmental Statistics*, **26(2)**:131 - 160.
> [10.1007/s13253-021-00447-1](https://doi.org/10.1007/s13253-021-00447-1)

## Quick start

The following code assume that 

## Set-up data for LDA

We set-up data for LDA from phyloseq object, the number of topics, hyper parameters for topic proportion in each sample and ASVs proportion in each topic.

We set the hyper-parameters $\alpha$ and $\gamma$ less than one to generate mixtures that are different from each other. We choose 0.8 for $\alpha$ across all samples so that we avoid generating unrealistic topics. 

```{r set-up, eval=FALSE}
K <- 11
stan.data <- setUpData(
  ps = psE_BARBI
  K = K,
  alpha = 0.9,
  gamma = 0.5
)
```

## Posterior sampling 

We need to estimate $\theta$ and $B = \left[\beta_{1}, \cdots, \beta_{K} \right]^{T}.$ 

We estimate the parameters using HMC NUTS with four chains and 2000 iterations. Out of these 2000 iterations, 1000 iterations were used as warmup samples. 

Here we want to save the posterior sampling results to
```{r file-name, eval=FALSE}
fileN <- paste0(
  "JABES_Endo_all_K_",
  K,
  "_ite_",
  iter,
  ".RData"
  )

```

Here we specify the LDA model written in lda.stan. We use `stan` function from `rstan` package. We save the results. We can do this in high-performance computing. 

```{r posterior-sampling, eval=FALSE}
stan.fit <- stan(
  file = "./lda.stan", 
  data = stan.data, 
  iter = iter, 
  chains = 4, 
  sample_file = NULL,
  diagnostic_file = NULL,
  cores = 4,
  control = list(adapt_delta = 0.9),
  save_dso = TRUE,
  algorithm = "NUTS"
  )
save(stan.fit, file = fileN)
```

## Alignment

We estimated the parameters using the HMC-NUTS sampler with four chains and 2000 iterations. Out of these 2000 iterations, 1000 iterations were used as warmup samples and discarded. Label switching across chains makes it difficult to directly compute log predictive density, split-$\hat{R}$, effective sample
size for model assessment, and evaluate convergence and mixing of chains.

To address this issue, we fixed the order of topics in chain one and then found the permutation that best aligned the topics across all four chains. For each chain two to four, we identified the estimated topics pair with the highest correlation, then found the next highest pair among the remaining, and so forth.

We extract posterior samples from the results.

```{r extract-ps-sample, eval=FALSE}
samples <- rstan::extract(
  stan.fit, 
  permuted = TRUE, 
  inc_warmup = FALSE, 
  include = TRUE
  )
```

We create a Topic $*$ Chain matrix.

```{r alignment, eval=FALSE}
theta <- samples$theta 
# dim(theta)
aligned <- alignmentMatrix(
  theta, 
  ps, 
  K, 
  iter = iter,
  chain = 4,
  SampleID_name = "unique_names"
  )

```

## Align topic proportion
We align the topic proportion in each specimen across four chains.

```{r align-theta, eval=FALSE}
theta_aligned <- thetaAligned(
  theta, 
  K, 
  aligned, 
  iter = iter, 
  chain = 4
  )
```
 
## Plot topic proportion

Plot the topic proportion in specimens and we can save it for puplication purposes. We can draw an informative summary of bacterial communities in each phenotype.

```{r plot-topic-prop, eval=FALSE}
p <- plotTopicProportion(
  ps,
  theta_aligned,
  K,
  col_names_theta_all = c("iteration", "Sample", "Topic", "topic.dis"),
  chain = 4,
  warmup = iter/2,
  SampleIDVarInphyloseq = "unique_names",
  design = ~ pna
)

ggsave(paste0("topic_dis_all_",K, ".png")), p, width = 30, height = 16)
```

## Align ASV proportion

We align the ASVs proportion in each topic across four chains.
```{r align-beta, eval=FALSE}
# an array (iterations *topic * ASV)
beta <- samples$beta 

# an array (iterations *topic * ASV)
beta_aligned <- betaAligned(
  beta, 
  K, 
  aligned, 
  iter = iter, 
  chain = 4
  ) 
```

## Plot ASVs distribution

We plot the ASVs distribution in each topic.

```{r ASV-prop, eval=FALSE}
p <- plotASVCircular(
  ps,
  K,
  iter = 2000,
  chain = 4,
  beta_aligned,
  varnames = c("iterations", "topic", "rsv_ix"),
  value_name = "beta_h",
  taxonomylevel = "Class",
  thresholdASVprop = 0.008
  )

ggsave(paste0("circular_plot_plant_all_K_", K, ".png"), p, width = 28, height = 22)
```

## Model assessement

```{r model-ass, eval=FALSE}
p_hist <- modelAssessment(
  ps,
  stan.fit,
  iter,
  statistic = "max",
  ASVsIndexToPlot = c(1, 3, 10:14, 19:26, 36, 51:53, 148)
)
ggsave(
  paste0(
    "model_assesment_hist_all_obs_sim_K_", 
    K, 
    ".png"
    ), 
  p_hist, 
  width = 12, 
  height = 6
  )
```


## Diagnostics - ESS
```{r diagnostics-ess, eval=FALSE}
p <- diagnosticsPlot(
  theta_aligned = theta_aligned,
  beta_aligned = beta_aligned
)
p_ess_bulk <- p[[[1]]]

ggsave(
  paste0("ess_bulk_plant_all_K_", K, ".png"), 
  p_ess_bulk, 
  width = 9, 
  height = 6
  )
```

## Diagnostics - $\hat{R}$
```{r diagnostics-rhat, eval=FALSE}

p_rhat <- p[[2]]

ggsave(
  paste0("Rhat_plant_all_K_", K, ".png"), 
  p_rhat, 
  width = 9, 
  height = 6
  )
```

## Differential topic analysis

### Test on all specimens

```{r all-specimens, eval=FALSE}
dds_all <- diffTop(
  design = ~ pna,
  ps,
  theta_aligned,
  subsetSample = sample_names(psE_BARBI)
)
res_all <- results(dds_all)
res_all
writeResTable(res_all, fileN = paste0("dds_all_K_", K,".tex"))
```


### Test on paired-specimens

```{r paired-specimens, eval=FALSE}
dds_all_paired <- diffTop(
  design = ~ pna,
  ps,
  theta_aligned,
  subsetSample = paired_specimens
)
res_paired <- results(dds_all_paired)
res_paired

writeResTable(res_paired, fileN = paste0("dds_all_paired_K_", K,".tex"))
```

### Test on paired non-Aster specimens

```{r non-aster-paired-specimens, eval=FALSE}
dds_all_non_aster_paired <- diffTop(
  design = ~ pna,
  ps,
  theta_aligned,
  subsetSample = paired_non_aster
)
res_paired_non_aster <- results(dds_all_non_aster_paired)
res_paired

writeResTable(res_paired_non_aster, fileN = paste0("dds_all_paired_nonaster_", K,".tex"))
```


### Test on paired Aster specimens

```{r aster-paired-specimens, eval=FALSE}
dds_all_aster_paired <- diffTop(
  design = ~ pna,
  ps,
  theta_aligned,
  subsetSample = paired_aster
)
res_paired_aster <- results(dds_all_aster_paired)
res_paired_aster

writeResTable(res_paired_aster, fileN = paste0("dds_all_paired_aster_K_", K,".rds"))
```
