<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Section 3: EXAMPLE OF 16S rRNA GENE SEQUENCING DATA • diffTop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Section 3: EXAMPLE OF 16S rRNA GENE SEQUENCING DATA">
<meta property="og:description" content="diffTop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">diffTop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../articles/diffTop.html">Get started</a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/appendix/01_introduction.html">Section 1: INTRODUCTION</a>
    </li>
    <li>
      <a href="../../articles/appendix/02_microbiome_data.html">Section 2: MICROBIOME DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/03_example_16S_data.html">Section 3: EXAMPLE OF 16S rRNA GENE SEQUENCING DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/04_models.html">Section 4: MODELS</a>
    </li>
    <li>
      <a href="../../articles/appendix/05_transformation.html">Section 5: TRANSFORMATIONS</a>
    </li>
    <li>
      <a href="../../articles/appendix/06_visualization.html">Section 6: VISUALIZATIONS FOR HETEROGENEOUS DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/07_multivariate_and_network_analysis.html">Section 7: MULTIVARIATE AND NETWORK ANALYSES</a>
    </li>
    <li>
      <a href="../../articles/appendix/08_differential_abundance_analysis.html">Section 8: DIFFERENTIAL ABUNDANCE ANALYSIS</a>
    </li>
    <li>
      <a href="../../articles/appendix/09_latent_Dirichlet_allocation.html">Section 9: LATENT DIRICHLET ALLOCATION</a>
    </li>
    <li>
      <a href="../../articles/appendix/10_uncertainity_quan_ordination.html">Section 10: UNCERTAINTY QUANTIFICATION FOR ORDINATION ANALYSIS</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="03_example_16S_data_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Section 3: EXAMPLE OF 16S rRNA GENE SEQUENCING DATA</h1>
                        <h4 class="author">Pratheepa Jeganathan</h4>
            
            <h4 class="date">03 August, 2021</h4>
      
      
      <div class="hidden name"><code>03_example_16S_data.Rmd</code></div>

    </div>

    
    
<p>We used published data to identify statistical challenges in analyzing molecular microbial data.</p>
<p>Using the <a href="https://datasetsearch.research.google.com/search?query=16S%20rRNA%20ASV%20Fitzpatrick&amp;docid=BifcGBcKEjlSM5VjAAAAAA%3D%3D">Google data set search</a>, we identified a dataset of 16S rRNA gene sequencing of the root endosphere specimens from Fitzpatrick et al. (2018).</p>
<p>In this section we create a <code>phyloseq</code> object from the <a href="https://figshare.com/articles/Chloroplast_sequence_variation_and_the_efficacy_of_peptide_nucleic_acids_for_blocking_host_amplification_in_plant_microbiome_studies/5948419/1">data downloaded from the website.</a></p>
<p>Then, we remove low coverage host plant type samples, removes Archaea, Eukaryota or unknown, remove unkown Phylum, remove mitochondira and Chloroplast.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/load_all.html">load_all</a></span><span class="op">(</span><span class="st">"."</span><span class="op">)</span></code></pre></div>
<div id="create-phyloseq-object" class="section level1">
<h1 class="hasAnchor">
<a href="#create-phyloseq-object" class="anchor"></a>Create phyloseq object</h1>
<p>We downloaded data from this <a href="https://figshare.com/articles/Chloroplast_sequence_variation_and_the_efficacy_of_peptide_nucleic_acids_for_blocking_host_amplification_in_plant_microbiome_studies/5948419/1">site</a></p>
<p>We have the following components.</p>
<ol style="list-style-type: decimal">
<li><p>PNA_root_microbiome_seqtab.rds - count matrix.</p></li>
<li><p>PNA_root_microbiome_SILVA_taxa.rds - taxonomy table.</p></li>
<li><p>PNA_metadata.csv - sample data.</p></li>
<li><p>PNA_16S_root_microbiome.tre - phylogenetic tree.</p></li>
</ol>
<p>Now, we create a <code>phyloseq</code> object.</p>
<p>Because the data set is large, we only show the R commands to create a <code>phyloseq</code> object. Then, we do processing of data and use that phyloseq for the downstream analysis. In <code>diffTop</code> package, we have the phylsoeq object with 6929 ASVs in specimens and controls.</p>
<p>Read all four files.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>
  <span class="st">"../data/PNA_root_microbiome_seqtab.rds"</span>
  <span class="op">)</span>
<span class="va">SILVA_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>
  <span class="st">"../data/PNA_root_microbiome_SILVA_taxa.rds"</span>
  <span class="op">)</span>
<span class="va">sam_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span>
  <span class="st">"../data/PNA_metadata.csv"</span>
  <span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/read_tree.html">read_tree</a></span><span class="op">(</span>
  <span class="st">"../data/PNA_16S_root_microbiome.tre"</span>
  <span class="op">)</span></code></pre></div>
<p>We have specimens in the rows of count matrix. We make sure the row names of sample data and the row names of the count matrix are the same.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sam_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span>
  <span class="va">seq_data</span>
  <span class="op">)</span><span class="co"># this is not the right thing to do, but the authors put the sample order in seq_data and the sam_data as same.</span></code></pre></div>
<p>We create phyloseq object and set specimens are in the columns of count matrix. We also name the identified taxonomy levels.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html">phyloseq</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span>
    <span class="va">seq_data</span>,
    taxa_are_rows <span class="op">=</span> <span class="cn">FALSE</span>
    <span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">sam_data</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">SILVA_taxa</span><span class="op">)</span>, 
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phy_tree-methods.html">phy_tree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">!=</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ntaxa-methods.html">ntaxa</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transpose-methods.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">ps</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="st">"Domain"</span>, 
  <span class="st">"Phylum"</span>, 
  <span class="st">"Class"</span>, 
  <span class="st">"Order"</span>, 
  <span class="st">"Family"</span>, 
  <span class="st">"Genus"</span>
  <span class="op">)</span></code></pre></div>
<p>After creating a phyloseq object we removed all four files.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">seq_data</span>, <span class="va">SILVA_taxa</span>, <span class="va">tree</span>, <span class="va">sam_data</span><span class="op">)</span></code></pre></div>
</div>
<div id="preprocessing" class="section level1">
<h1 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>Preprocessing</h1>
<ul>
<li>Remove low coverage host plant type samples.</li>
<li>Removes Archaea, Eukaryota or unknown.</li>
<li>Remove unkown Phylum.</li>
<li>Remove mitochondira and Chloroplast.</li>
</ul>
<p>Remove low coverage host plant type samples with less than 3 replicates, but we keep plant type RX. Here species refer to plant type.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">less_num_replicates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">species</span>
    <span class="op">)</span>
  <span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">species</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">3</span><span class="op">)</span><span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">species</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">species</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">species</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"control"</span> 

<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">species</span>
  <span class="op">)</span>
<span class="co"># from the above list we didn't want to drop plant type "RX"</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  <span class="va">species</span> <span class="op">!=</span> <span class="st">"131"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"139"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"14"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"146"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"8"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"93"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"94"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"149"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"154"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"161"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"164"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"56"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"79"</span> <span class="op">&amp;</span> <span class="va">species</span> <span class="op">!=</span> <span class="st">"99"</span>
  <span class="op">)</span>
<span class="va">ps</span></code></pre></div>
<p>Removes Archaea, Eukaryota or unknown.</p>
<p>We removed ASVs that lacked a kingdom assignment or were assigned to Archaea or Eukaryota (3597 ASVs).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  <span class="va">Domain</span> <span class="op">==</span> <span class="st">"Bacteria"</span>
  <span class="op">)</span> 
<span class="va">ps</span></code></pre></div>
<p>Remove unkown Phylum.</p>
<p>We removed ASVs that lacked a phylum assignment (3521 ASVs).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">Phylum</span><span class="op">)</span>
  <span class="op">)</span>
<span class="va">ps</span></code></pre></div>
<p>Remove mitochondira and Chloroplast</p>
<p>We removed ASVs classified as mitochondria.</p>
<p>Mitochondria is a contamination. We can identify mitochondria using it’s family name (removing 1417 ASVs).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  <span class="op">!</span><span class="va">Family</span> <span class="op">%in%</span> <span class="st">'Mitochondria'</span>
  <span class="op">)</span> 
<span class="va">ps</span> 

<span class="co">#calculate # of usable reads after removing above but before removing chloroplast</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">Reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span></code></pre></div>
<p>We removed ASVs classified as Chloroplast</p>
<p>Chloroplast is a contamination. We can identify chloroplast using it’s class name (removing 481 ASVs).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span>
  <span class="va">ps</span>, <span class="op">!</span><span class="va">Class</span> <span class="op">%in%</span> <span class="st">"Chloroplast"</span>
  <span class="op">)</span> 
<span class="va">ps</span> 

<span class="co">#calculate # of usable reads after removing chloroplast</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">ReadsNC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html">sample_sums</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span></code></pre></div>
<p>Add plant type names, Asteraceae or non-Asteraceae to the phyloseq sample data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"144"</span>,<span class="st">"105"</span>,<span class="st">"106"</span>,<span class="st">"66"</span>,<span class="st">"30"</span>,<span class="st">"20"</span>,<span class="st">"112"</span>,<span class="st">"72"</span>,<span class="st">"71"</span>,<span class="st">"162"</span>,<span class="st">"95"</span>,<span class="st">"92"</span>,<span class="st">"54"</span>,<span class="st">"RX"</span>,<span class="st">"100"</span>,<span class="st">"101"</span>,<span class="st">"134"</span>,<span class="st">"87"</span>,<span class="st">"33"</span>,<span class="st">"137"</span>,<span class="st">"45"</span>,<span class="st">"108"</span>,<span class="st">"109"</span>,<span class="st">"17"</span>,<span class="st">"22"</span>,<span class="st">"142"</span>,<span class="st">"143"</span>,<span class="st">"40"</span>,<span class="st">"ST"</span>,<span class="st">"15"</span>,<span class="st">"113"</span>,<span class="st">"7"</span><span class="op">)</span>,
  species_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Sporobolus crytptandrus"</span>, <span class="st">"Phalaris arundinacea"</span>, <span class="st">"Phleum pratense"</span>, <span class="st">"Festuca arundinacea"</span>, <span class="st">"Bromus inermis"</span>, <span class="st">"Asparagus officinalis"</span>, <span class="st">"Potentilla recta"</span>, <span class="st">"Geum canadense"</span>,<span class="st">"Geum aleppicum"</span>, <span class="st">"Vicia tetrasperma"</span>, <span class="st">"Medicago sativa"</span>, <span class="st">"Lotus corniculatus"</span>, <span class="st">"Desmodium canadense"</span>, <span class="st">"Rhexia virginica"</span>, <span class="st">"Oenothera biennis"</span>, <span class="st">"Oenothera perennis"</span>, <span class="st">"Sisymbrium officinale"</span>, <span class="st">"Lepidium densiflorum"</span>, <span class="st">"Capsella bursa-pstoris"</span>, <span class="st">"Solanum dulcamara"</span>, <span class="st">"Convolvulus arvensis"</span>, <span class="st">"Plantago major"</span>, <span class="st">"Plantago rugelii"</span>, <span class="st">"Asclepias incarnata"</span>, <span class="st">"Symphyotrichum ericoides"</span>, <span class="st">"Sonchus arvensis"</span>, <span class="st">"Sonchus oleraceus"</span>, <span class="st">"Cichorium intybus"</span>, <span class="st">"Centaurea solstitialis"</span>, <span class="st">"Arctium minus"</span>, <span class="st">"Persicaria maculosa"</span>, <span class="st">"Amaranthus albus"</span><span class="op">)</span>,
  Type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"non-aster"</span>, <span class="fl">24</span><span class="op">)</span>, 
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"aster"</span>, <span class="fl">6</span><span class="op">)</span>, 
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"non-aster"</span>, <span class="fl">2</span><span class="op">)</span>
    <span class="op">)</span>
  <span class="op">)</span>

<span class="co">## add the above info to sample data</span>
<span class="va">sam</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">sam</span><span class="op">$</span><span class="va">unique_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sam</span><span class="op">)</span>
<span class="va">sam_new</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span>
  <span class="va">sam</span>, 
  <span class="va">df</span>, 
  by <span class="op">=</span> <span class="st">"species"</span>
  <span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sam_new</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam_new</span><span class="op">$</span><span class="va">unique_names</span>
<span class="va">sam_new</span> <span class="op">&lt;-</span> <span class="va">sam_new</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sam</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_phyloseq.html">merge_phyloseq</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>, 
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">sam_new</span><span class="op">)</span>, 
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phy_tree-methods.html">phy_tree</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>
  <span class="op">)</span>
<span class="va">ps</span></code></pre></div>
<p>We kept only plant types that have endosphere or rhizosphere specimens.</p>
<p>We removed samples with less than 800 reads (including contaminant reads of mitochondira and chloroplast). We saved the reads in the sample information before removing contaminants (mitochondira and chloroplast).</p>
<p>We removed samples from sequencing run == 5 based on the filtering in Fitzpatrick et al. (2018).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># keep only plant types with Endosphere and Rhizosphere </span>
<span class="va">psER</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  <span class="va">species</span> <span class="op">==</span> <span class="st">"15"</span> <span class="op">|</span> <span class="va">species</span> <span class="op">==</span> <span class="st">"22"</span> <span class="op">|</span> <span class="va">species</span> <span class="op">==</span> <span class="st">"142"</span> <span class="op">|</span> <span class="va">species</span> <span class="op">==</span> <span class="st">"40"</span> <span class="op">|</span> <span class="va">species</span> <span class="op">==</span> <span class="st">"143"</span> <span class="op">|</span> <span class="va">species</span> <span class="op">==</span> <span class="st">"ST"</span> <span class="op">|</span>  <span class="va">species</span> <span class="op">==</span> <span class="st">"33"</span> <span class="op">|</span> <span class="va">species</span> <span class="op">==</span> <span class="st">"71"</span> <span class="op">|</span> <span class="va">species</span> <span class="op">==</span> <span class="st">"106"</span><span class="op">|</span><span class="va">species</span> <span class="op">==</span> <span class="st">"control"</span>
  <span class="op">)</span>

<span class="va">psER</span>

<span class="va">psER</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span>
  <span class="va">psER</span>, 
  <span class="va">Reads</span> <span class="op">&gt;=</span> <span class="fl">800</span>
  <span class="op">)</span> 
<span class="va">psER</span> <span class="co"># removes 7 plant samples</span>

<span class="va">psER</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span>
  <span class="va">psER</span>, 
  <span class="va">run</span> <span class="op">&lt;</span> <span class="fl">5</span> <span class="op">|</span> <span class="va">run</span> <span class="op">&gt;</span> <span class="fl">5</span>
  <span class="op">)</span> <span class="co"># remove samples from Run 5</span>
<span class="va">psER</span></code></pre></div>
<p>We Kept Endosphere samples (root, leaves or any part of plant).</p>
<p>We removed samples with less than 800 reads (non contaminant reads).</p>
<p>We removed C. solstitalis leave samples.</p>
<p>We removed ASV with no reads.</p>
<p>We removed ASVs with less than 25 reads in at most two samples.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># remove rhizosphere samples </span>
<span class="va">psE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span>
  <span class="va">psER</span>, 
  <span class="va">community</span> <span class="op">==</span> <span class="st">"E"</span> <span class="op">|</span> <span class="va">species</span> <span class="op">==</span> <span class="st">"control"</span>
  <span class="op">)</span>
<span class="va">psE</span>

<span class="va">psE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span>
  <span class="va">psE</span>, 
  <span class="va">ReadsNC</span> <span class="op">&gt;=</span> <span class="fl">800</span>
  <span class="op">)</span>
<span class="va">psE</span>

<span class="co"># remove C. solstitalis leave samples</span>
<span class="va">leaves</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="st">"ST-HU01-8-L-M"</span>,
  <span class="st">"ST-HU01-8-L-O"</span>
  <span class="op">)</span>

<span class="va">psE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span>
  <span class="va">psE</span>, 
  <span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">psE</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">leaves</span>
  <span class="op">)</span>
<span class="va">psE</span>

<span class="va">psE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">psE</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, 
  <span class="va">psE</span>
  <span class="op">)</span>
<span class="va">psE</span> </code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"psE"</span><span class="op">)</span>
<span class="va">psE</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 6929 taxa and 111 samples ]
## sample_data() Sample Data:       [ 111 samples by 13 sample variables ]
## tax_table()   Taxonomy Table:    [ 6929 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 6929 tips and 6928 internal nodes ]</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pratheepa Jeganathan, Susan Holmes.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
