<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Section 8: DIFFERENTIAL ABUNDANCE ANALYSIS â€¢ diffTop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Section 8: DIFFERENTIAL ABUNDANCE ANALYSIS">
<meta property="og:description" content="diffTop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">diffTop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/appendix/01_introduction.html">Section 1: INTRODUCTION</a>
    </li>
    <li>
      <a href="../../articles/appendix/02_microbiome_data.html">Section 2: MICROBIOME DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/03_example_16S_data.html">Section 3: EXAMPLE OF 16S rRNA GENE SEQUENCING DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/04_models.html">Section 4: MODELS</a>
    </li>
    <li>
      <a href="../../articles/appendix/05_transformation.html">Section 5: TRANSFORMATIONS</a>
    </li>
    <li>
      <a href="../../articles/appendix/06_visualization.html">Section 6: VISUALIZATIONS FOR HETEROGENEOUS DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/07_multivariate_and_network_analysis.html">Section 7: MULTIVARIATE AND NETWORK ANALYSES</a>
    </li>
    <li>
      <a href="../../articles/appendix/08_differential_abundance_analysis.html">Section 8: DIFFERENTIAL ABUNDANCE ANALYSIS</a>
    </li>
    <li>
      <a href="../../articles/appendix/09_latent_Dirichlet_allocation.html">Section 9: LATENT DIRICHLET ALLOCATION</a>
    </li>
    <li>
      <a href="../../articles/appendix/10_uncertainity_quan_ordination.html">Section 10: UNCERTAINTY QUANTIFICATION FOR ORDINATION ANALYSIS</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="08_differential_abundance_analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="08_differential_abundance_analysis_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="08_differential_abundance_analysis_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Section 8: DIFFERENTIAL ABUNDANCE ANALYSIS</h1>
                        <h4 class="author">Pratheepa Jeganathan and Susan Holmes</h4>
            
            <h4 class="date">23 February, 2021</h4>
      
      
      <div class="hidden name"><code>08_differential_abundance_analysis.Rmd</code></div>

    </div>

    
    
<p>An important goal in microbiome research is often to find taxonomic differences across environments or groups.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">genefilter</span><span class="op">)</span> <span class="co">#KOverA</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://cran.r-project.org">vegan</a></span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/load_all.html">load_all</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">theme_set</span><span class="op">(</span><span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu">theme_update</span><span class="op">(</span>
  text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
  legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<div id="data" class="section level2">
<h2 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">threshold</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genefilter/man/kOverA.html">kOverA</a></span><span class="op">(</span><span class="fl">2</span>, A <span class="op">=</span> <span class="fl">25</span><span class="op">)</span> 
<span class="va">psE_BARBI</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/filter_taxa.html">filter_taxa</a></span><span class="op">(</span>
  <span class="va">psE_BARBI</span>, 
  <span class="va">threshold</span>, <span class="cn">TRUE</span><span class="op">)</span> 

<span class="va">psE_BARBI</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1418 taxa and 86 samples ]
## sample_data() Sample Data:       [ 86 samples by 16 sample variables ]
## tax_table()   Taxonomy Table:    [ 1418 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1418 tips and 1417 internal nodes ]</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">psE_BARBI</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">psE_BARBI</span><span class="op">)</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">ps</span><span class="op">)</span>
<span class="va">ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1418 taxa and 86 samples ]
## sample_data() Sample Data:       [ 86 samples by 16 sample variables ]
## tax_table()   Taxonomy Table:    [ 1418 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1418 tips and 1417 internal nodes ]</code></pre>
</div>
<div id="permutation-tests-using-distance-matrices" class="section level2">
<h2 class="hasAnchor">
<a href="#permutation-tests-using-distance-matrices" class="anchor"></a>8.1 Permutation tests using distance matrices</h2>
<div id="power-calculation-with-and-without-strain-switching" class="section level3">
<h3 class="hasAnchor">
<a href="#power-calculation-with-and-without-strain-switching" class="anchor"></a>Power calculation with and without strain switching</h3>
<p>One difficulty with current 16S rRNA data are that the distance based tests are very sensitive to the choice of distance and the presence of strain switching can substantially decrease the power of the test.</p>
<p>We show that for the exemplary data this is in fact the case as strains ASV 153 is switched with ASVs 12, 354 and 345. To illustrate the decrease in power in PERMANOVA through a simulation, we generated negative binomial count data with parameters similar to these ASVs and show that when a species switches ASV, ie is present as one ASV in one set of samples and a close, distinct strain appears in the other set of samples, the power to detect a difference using a Bray Curtis distance and permanova is considerably diminished.</p>
</div>
<div id="a-function-to-generate-the-data-with-nb-marginals" class="section level3">
<h3 class="hasAnchor">
<a href="#a-function-to-generate-the-data-with-nb-marginals" class="anchor"></a>A function to generate the data with NB marginals</h3>
<ul>
<li><p>With strain switching (split=TRUE) for k ASVs or</p></li>
<li><p>Without strain switching (split=FALSE).</p></li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">generatematasv</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span>
  <span class="va">n</span><span class="op">=</span><span class="fl">100</span>, 
  <span class="va">k</span><span class="op">=</span><span class="fl">1</span>, 
  <span class="va">split</span><span class="op">=</span><span class="cn">TRUE</span>,
  <span class="va">p</span><span class="op">=</span><span class="fl">20</span>, 
  <span class="va">n1</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">floor</a></span><span class="op">(</span><span class="va">n</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>, 
  <span class="va">n2</span><span class="op">=</span><span class="va">n</span><span class="op">-</span><span class="va">n1</span>, 
  <span class="va">group1</span><span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="va">n1</span>, 
  <span class="va">diff</span><span class="op">=</span><span class="fl">4</span>, 
  <span class="va">mu0</span><span class="op">=</span><span class="fl">100</span>, 
  <span class="va">size0</span><span class="op">=</span><span class="fl">0.2</span>
  <span class="op">)</span><span class="op">{</span>
    <span class="co">## Different means version, in vars 1:k</span>
    <span class="co">## split is TRUE when there is strain  switching</span>
    <span class="co">## The number of ASVs is 2p</span>
    <span class="co">## k identifies how many ASVs are changed across groups  </span>
    <span class="co">## n1 is the size of group 1</span>
    <span class="co">## n2 is the size of group 2</span>
    <span class="co">## diff is the multiplicative effect size difference  </span>
    <span class="va">matasv</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">0</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">*</span><span class="va">p</span>, nrow<span class="op">=</span><span class="va">n</span><span class="op">)</span>
    <span class="va">asv12</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">n</span><span class="op">)</span>
    <span class="kw">if</span> <span class="op">(</span><span class="va">k</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span><span class="op">{</span>
        <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">k</span><span class="op">)</span><span class="op">{</span>
          <span class="va">asv12</span><span class="op">[</span><span class="va">group1</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html">rnbinom</a></span><span class="op">(</span><span class="va">n1</span>, size<span class="op">=</span><span class="va">size0</span>, mu<span class="op">=</span><span class="va">diff</span><span class="op">*</span><span class="va">mu0</span><span class="op">)</span>  
          <span class="va">asv12</span><span class="op">[</span><span class="op">-</span><span class="va">group1</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html">rnbinom</a></span><span class="op">(</span><span class="va">n2</span>, size<span class="op">=</span><span class="va">size0</span>, mu<span class="op">=</span><span class="va">mu0</span><span class="op">)</span>  
          <span class="co"># no switching</span>
          <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="va">split</span><span class="op">)</span><span class="op">{</span>
            <span class="co">#One otus that has a difference  </span>
            <span class="va">matasv</span><span class="op">[</span>,<span class="fl">1</span><span class="op">+</span><span class="fl">2</span><span class="op">*</span><span class="op">(</span><span class="va">j</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span> <span class="va">asv12</span>
            <span class="co">#One otu with no difference</span>
            <span class="va">matasv</span><span class="op">[</span>,<span class="fl">2</span><span class="op">*</span><span class="va">j</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html">rnbinom</a></span><span class="op">(</span><span class="va">n</span>,size<span class="op">=</span><span class="va">size0</span>,mu<span class="op">=</span><span class="va">mu0</span><span class="op">)</span>
          <span class="op">}</span>
          <span class="kw">if</span><span class="op">(</span><span class="va">split</span><span class="op">)</span><span class="op">{</span> 
            <span class="co">#If there is strain switching</span>
            <span class="co"># one NB is split across two columns</span>
            <span class="co"># Alternate where the strains occr</span>
            <span class="co"># in the vector of indices ind</span>
            <span class="va">ind</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="va">n</span>,by<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
            <span class="va">matasv</span><span class="op">[</span><span class="va">ind</span>,<span class="fl">1</span><span class="op">+</span><span class="fl">2</span><span class="op">*</span><span class="op">(</span><span class="va">j</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span><span class="va">asv12</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span>
            <span class="va">matasv</span><span class="op">[</span><span class="op">-</span><span class="va">ind</span>,<span class="fl">2</span><span class="op">*</span><span class="va">j</span><span class="op">]</span><span class="op">&lt;-</span><span class="va">asv12</span><span class="op">[</span><span class="op">-</span><span class="va">ind</span><span class="op">]</span>
            <span class="op">}</span>
        <span class="op">}</span>
        
        <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="op">(</span><span class="va">k</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">:</span><span class="va">p</span><span class="op">)</span><span class="op">{</span>
          <span class="co">##the rest are unchanged between groups</span>
          <span class="va">matasv</span><span class="op">[</span>,<span class="fl">1</span><span class="op">+</span><span class="fl">2</span><span class="op">*</span><span class="op">(</span><span class="va">j</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html">rnbinom</a></span><span class="op">(</span><span class="va">n</span>,size<span class="op">=</span><span class="va">size0</span>,mu<span class="op">=</span><span class="va">mu0</span><span class="op">)</span>
          <span class="va">matasv</span><span class="op">[</span>,<span class="fl">2</span><span class="op">*</span><span class="va">j</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html">rnbinom</a></span><span class="op">(</span><span class="va">n</span>,size<span class="op">=</span><span class="va">size0</span>,mu<span class="op">=</span><span class="va">mu0</span><span class="op">)</span>
        <span class="op">}</span>
    <span class="op">}</span>  
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">matasv</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div id="permanova-in-adonis-package" class="section level3">
<h3 class="hasAnchor">
<a href="#permanova-in-adonis-package" class="anchor"></a>PERMANOVA in adonis package</h3>
<p>A standard <code>permanova</code> test can be run using the <code>adonis</code> command from the <code>vegan</code> package.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">n0</span><span class="op">&lt;-</span><span class="fl">50</span>
<span class="va">n1</span><span class="op">&lt;-</span><span class="fl">25</span>;<span class="va">n2</span><span class="op">&lt;-</span><span class="fl">25</span>
<span class="va">matasv</span><span class="op">&lt;-</span><span class="fu">generatematasv</span><span class="op">(</span>n<span class="op">=</span><span class="va">n0</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">matasv</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 50 40</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">##matasv is generated as a </span>
<span class="co">##multivariate with negative binomial marginals</span>

<span class="va">afact</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="va">n1</span><span class="op">)</span>, 
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"B"</span>,<span class="va">n2</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span>

<span class="va">distasv</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html">vegdist</a></span><span class="op">(</span><span class="va">matasv</span>, method<span class="op">=</span><span class="st">"bray"</span><span class="op">)</span>
<span class="va">ptest1</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/adonis.html">adonis</a></span><span class="op">(</span><span class="va">distasv</span><span class="op">~</span><span class="va">afact</span><span class="op">)</span>
<span class="va">ptest1</span></code></pre></div>
<pre><code>## 
## Call:
## adonis(formula = distasv ~ afact) 
## 
## Permutation: free
## Number of permutations: 999
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2 Pr(&gt;F)
## afact      1    0.3465 0.34654  1.0671 0.02175  0.388
## Residuals 48   15.5887 0.32476         0.97825       
## Total     49   15.9352                 1.00000</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#To extract the pvalue:</span>
<span class="va">ptest1</span><span class="op">$</span><span class="va">aov.tab</span><span class="op">$</span><span class="va">`Pr(&gt;F)`</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></code></pre></div>
<pre><code>## [1] 0.388</code></pre>
</div>
<div id="power-computation-with-a-difference-in-non-switched-standard-and-switched-otus" class="section level3">
<h3 class="hasAnchor">
<a href="#power-computation-with-a-difference-in-non-switched-standard-and-switched-otus" class="anchor"></a>Power computation with a difference in non switched (standard) and switched OTUs</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">n0</span><span class="op">&lt;-</span><span class="fl">100</span>
<span class="va">poweranalysis</span><span class="op">&lt;-</span><span class="kw">function</span><span class="op">(</span>
  <span class="va">B</span><span class="op">=</span><span class="fl">1000</span>,
  <span class="va">diff0</span><span class="op">=</span><span class="fl">4</span>,
  <span class="va">k0</span><span class="op">=</span><span class="fl">1</span>,
  <span class="va">nbASVs</span><span class="op">=</span><span class="fl">40</span>,
  <span class="va">n</span><span class="op">=</span><span class="va">n0</span>,
  <span class="va">split0</span> <span class="op">=</span> <span class="cn">FALSE</span>,
  <span class="va">mu0</span><span class="op">=</span><span class="fl">500</span>,
  <span class="va">size0</span><span class="op">=</span><span class="fl">0.2</span><span class="op">)</span><span class="op">{</span>

    <span class="va">pvals</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">B</span><span class="op">)</span>
    <span class="va">afact</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>
      <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="va">n</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>,
        <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"B"</span>,<span class="va">n</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>
      <span class="op">)</span>
    <span class="kw">for</span> <span class="op">(</span><span class="va">b</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">B</span><span class="op">)</span><span class="op">{</span>
        <span class="va">matasv</span><span class="op">&lt;-</span><span class="fu">generatematasv</span><span class="op">(</span>
          n<span class="op">=</span><span class="va">n0</span>, 
          k<span class="op">=</span><span class="va">k0</span>, 
          p<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">nbASVs</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>,
          diff<span class="op">=</span><span class="va">diff0</span>,
          split <span class="op">=</span> <span class="va">split0</span>,
          mu0<span class="op">=</span><span class="va">mu0</span>,
          size0<span class="op">=</span><span class="va">size0</span>
          <span class="op">)</span>
        <span class="va">distasv</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html">vegdist</a></span><span class="op">(</span>
          <span class="va">matasv</span>,
          method<span class="op">=</span><span class="st">"bray"</span>
          <span class="op">)</span>
        
        <span class="va">ptest1</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/adonis.html">adonis</a></span><span class="op">(</span><span class="va">distasv</span><span class="op">~</span><span class="va">afact</span><span class="op">)</span>
        <span class="va">pvals</span><span class="op">[</span><span class="va">b</span><span class="op">]</span><span class="op">&lt;-</span><span class="va">ptest1</span><span class="op">$</span><span class="va">aov.tab</span><span class="op">$</span><span class="va">`Pr(&gt;F)`</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
    <span class="op">}</span>
    <span class="va">power005</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span>
    <span class="va">power001</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">&lt;</span><span class="fl">0.01</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">power005</span>,<span class="va">power001</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Here we use large mu0 and the size (similar to ASV 153)  </span>
<span class="va">powerswitch</span><span class="op">&lt;-</span><span class="fu">poweranalysis</span><span class="op">(</span>
  B<span class="op">=</span><span class="fl">1000</span>,
  diff0<span class="op">=</span><span class="fl">5</span>, 
  k0<span class="op">=</span><span class="fl">1</span>, 
  nbASVs<span class="op">=</span><span class="fl">20</span>,
  n<span class="op">=</span><span class="va">n0</span>, 
  split0<span class="op">=</span><span class="cn">TRUE</span>,
  mu0<span class="op">=</span><span class="fl">500</span>,
  size<span class="op">=</span><span class="fl">0.2</span>
  <span class="op">)</span>

<span class="va">powerstand</span><span class="op">&lt;-</span><span class="fu">poweranalysis</span><span class="op">(</span>
  B<span class="op">=</span><span class="fl">1000</span>,
  diff0<span class="op">=</span><span class="fl">5</span>, 
  k0<span class="op">=</span><span class="fl">1</span>, 
  nbASVs<span class="op">=</span><span class="fl">20</span>, 
  n<span class="op">=</span><span class="va">n0</span>, 
  split0<span class="op">=</span><span class="cn">FALSE</span>,
  mu0<span class="op">=</span><span class="fl">500</span>,
  size<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span>

<span class="fu">usethis</span><span class="fu">::</span><span class="fu"><a href="https://usethis.r-lib.org/reference/use_data.html">use_data</a></span><span class="op">(</span><span class="va">powerswitch</span><span class="op">)</span>
<span class="fu">usethis</span><span class="fu">::</span><span class="fu"><a href="https://usethis.r-lib.org/reference/use_data.html">use_data</a></span><span class="op">(</span><span class="va">powerstand</span><span class="op">)</span></code></pre></div>
<p>In the case of strain switching standard PERMANOVA will see itâ€™s power substantially decreased from 0.51 to 0.25.</p>
</div>
</div>
<div id="differential-abundance-through-generalized-linear-modeling-and-transformatios" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-abundance-through-generalized-linear-modeling-and-transformatios" class="anchor"></a>8.2 Differential abundance through generalized linear modeling and transformatios</h2>
<p>Simple two-sample testing is marred by several technological difficulties. There are batch effects, technical biases, and heteroscedasticity in the prevalence estimates due to differences in the library sizes across specimens as well as strain switching (where ASV strains are replaced as we change locations or subjects).</p>
</div>
<div id="communities-instead-of-individual-taxa" class="section level2">
<h2 class="hasAnchor">
<a href="#communities-instead-of-individual-taxa" class="anchor"></a>8.3 Communities instead of individual taxa</h2>
<p>Topic models provide useful aggregates that can be used for differential abundance analysis based on topics rather than individual strains.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pratheepa Jeganathan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
