<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Section 9: LATENT DIRICHLET ALLOCATION â€¢ diffTop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Section 9: LATENT DIRICHLET ALLOCATION">
<meta property="og:description" content="diffTop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">diffTop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../articles/diffTop.html">Get started</a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/appendix/01_introduction.html">Section 1: INTRODUCTION</a>
    </li>
    <li>
      <a href="../../articles/appendix/02_microbiome_data.html">Section 2: MICROBIOME DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/03_example_16S_data.html">Section 3: EXAMPLE OF 16S rRNA GENE SEQUENCING DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/04_models.html">Section 4: MODELS</a>
    </li>
    <li>
      <a href="../../articles/appendix/05_transformation.html">Section 5: TRANSFORMATIONS</a>
    </li>
    <li>
      <a href="../../articles/appendix/06_visualization.html">Section 6: VISUALIZATIONS FOR HETEROGENEOUS DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/07_multivariate_and_network_analysis.html">Section 7: MULTIVARIATE AND NETWORK ANALYSES</a>
    </li>
    <li>
      <a href="../../articles/appendix/08_differential_abundance_analysis.html">Section 8: DIFFERENTIAL ABUNDANCE ANALYSIS</a>
    </li>
    <li>
      <a href="../../articles/appendix/09_latent_Dirichlet_allocation.html">Section 9: LATENT DIRICHLET ALLOCATION</a>
    </li>
    <li>
      <a href="../../articles/appendix/10_uncertainity_quan_ordination.html">Section 10: UNCERTAINTY QUANTIFICATION FOR ORDINATION ANALYSIS</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="09_latent_Dirichlet_allocation_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Section 9: LATENT DIRICHLET ALLOCATION</h1>
                        <h4 class="author">Pratheepa Jeganathan</h4>
            
            <h4 class="date">04 August, 2021</h4>
      
      
      <div class="hidden name"><code>09_latent_Dirichlet_allocation.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">genefilter</span><span class="op">)</span> 
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.rcpp.org">Rcpp</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://mc-stan.org/rstan">rstan</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">randomcoloR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">diffTop</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">theme_set</span><span class="op">(</span><span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu">theme_update</span><span class="op">(</span>
  text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
  legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p>Here, we are interested infer on the topics (bacterial communities) when we use original and modified PNA clams (two different experimental conditions). We use latent Dirichlet allocation to identify topics.</p>
<div id="data" class="section level2">
<h2 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h2>
<p>In this workflow we use <code>psE_BARBI</code> <code>phyloseq</code> object in this package. In <code>psE_BARBI</code>, we removed DNA contaminants using BARBI.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"psE_BARBI"</span><span class="op">)</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">psE_BARBI</span>

<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">psE_BARBI</span><span class="op">)</span></code></pre></div>
<div id="preprocessing" class="section level3">
<h3 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>Preprocessing</h3>
<p>We choose ASVs that have at least 25 counts in at least two specimens.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">threshold</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genefilter/man/kOverA.html">kOverA</a></span><span class="op">(</span><span class="fl">2</span>, A <span class="op">=</span> <span class="fl">25</span><span class="op">)</span> 
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/filter_taxa.html">filter_taxa</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  <span class="va">threshold</span>, <span class="cn">TRUE</span><span class="op">)</span> </code></pre></div>
<pre><code>## Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'</code></pre>
<pre><code>## Also defined by 'tidytree'</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1418 taxa and 86 samples ]
## sample_data() Sample Data:       [ 86 samples by 16 sample variables ]
## tax_table()   Taxonomy Table:    [ 1418 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1418 tips and 1417 internal nodes ]</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">ps</span><span class="op">)</span></code></pre></div>
<pre><code>## Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
## Also defined by 'tidytree'</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1418 taxa and 86 samples ]
## sample_data() Sample Data:       [ 86 samples by 16 sample variables ]
## tax_table()   Taxonomy Table:    [ 1418 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1418 tips and 1417 internal nodes ]</code></pre>
</div>
<div id="edit-specimen-names" class="section level3">
<h3 class="hasAnchor">
<a href="#edit-specimen-names" class="anchor"></a>Edit specimen names</h3>
<p>We edit specimen names and identify Asteraceae and non-Asteraceae plants.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sam_names</span> <span class="op">&lt;-</span> <span class="fu">str_replace</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>, <span class="st">"E106"</span>, <span class="st">"E-106"</span><span class="op">)</span>
<span class="va">sam_names</span> <span class="op">&lt;-</span> <span class="fu">str_replace</span><span class="op">(</span><span class="va">sam_names</span>, <span class="st">"_F_filt.fastq.gz"</span>, <span class="st">""</span><span class="op">)</span>
<span class="va">sam_names</span> <span class="op">&lt;-</span> <span class="fu">str_replace</span><span class="op">(</span><span class="va">sam_names</span>, <span class="st">"Connor-"</span>, <span class="st">"E"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam_names</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">X</span> <span class="op">&lt;-</span> <span class="va">sam_names</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">unique_names</span> <span class="op">&lt;-</span> <span class="va">sam_names</span>
<span class="va">aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"142"</span>,<span class="st">"143"</span>,<span class="st">"15"</span>,<span class="st">"ST"</span>,<span class="st">"22"</span>,<span class="st">"40"</span><span class="op">)</span>
<span class="va">non_aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"33"</span>, <span class="st">"71"</span>, <span class="st">"106"</span><span class="op">)</span>

<span class="va">paired_aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"E-142-1"</span>, <span class="st">"E142-1"</span>, <span class="st">"E-142-5"</span>, <span class="st">"E142-5"</span>, <span class="st">"E-142-10"</span>, <span class="st">"E142-10"</span>, <span class="st">"E-143-2"</span>, <span class="st">"E143-2"</span>, <span class="st">"E-143-7"</span>, <span class="st">"E143-7"</span>, <span class="st">"E-15-1"</span>, <span class="st">"E15-1"</span>, <span class="st">"ST-CAZ-4-R-O"</span>, <span class="st">"ST-CAZ-4-R-M"</span>, <span class="st">"ST-SAL-22-R-O"</span>, <span class="st">"ST-SAL-22-R-M"</span>, <span class="st">"ST-TRI-10-R-O"</span>, <span class="st">"ST-TRI-10-R-M"</span><span class="op">)</span>
<span class="va">paired_non_aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"E33-7"</span>, <span class="st">"E-33-7"</span>, <span class="st">"E33-8"</span>, <span class="st">"E-33-8"</span>, <span class="st">"E33-9"</span>, <span class="st">"E-33-9"</span>, <span class="st">"E71-10"</span>, <span class="st">"E-71-10"</span>,<span class="st">"E71-2"</span>,<span class="st">"E-71-2"</span> ,<span class="st">"E71-3"</span>, <span class="st">"E-71-3"</span>, <span class="st">"E106-1"</span>, <span class="st">"E-106-1"</span>, <span class="st">"E106-3"</span>, <span class="st">"E-106-3"</span>, <span class="st">"E106-4"</span>, <span class="st">"E-106-4"</span><span class="op">)</span>
<span class="va">paired_specimens</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">paired_aster</span>, <span class="va">paired_non_aster</span><span class="op">)</span></code></pre></div>
<p>Choose colors for plots. We will use nine colors for nine different plants in the dataset.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plant_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#999999"</span>, <span class="st">"#E69F00"</span>, <span class="st">"#56B4E9"</span>, <span class="st">"#009E73"</span>, <span class="st">"#F0E442"</span>, <span class="st">"#0072B2"</span>, <span class="st">"#D55E00"</span>, <span class="st">"#CC79A7"</span>, <span class="st">"purple"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="input-data-for-lda" class="section level2">
<h2 class="hasAnchor">
<a href="#input-data-for-lda" class="anchor"></a>Input data for LDA</h2>
<p>Based on the ordination, we choose the number of topics as K=11. We set the hyperparameters <span class="math inline">\(\alpha = 0.8\)</span> and <span class="math inline">\(\gamma = 0.8\)</span>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">K</span> <span class="op">&lt;-</span> <span class="fl">11</span>
<span class="va">alpha</span> <span class="op">&lt;-</span> <span class="fl">0.8</span>
<span class="va">gamma</span> <span class="op">&lt;-</span> <span class="fl">0.8</span>
<span class="va">stan.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/setUpData.html">setUpData</a></span><span class="op">(</span>
  ps <span class="op">=</span> <span class="va">ps</span>,
  K <span class="op">=</span> <span class="va">K</span>,
  alpha <span class="op">=</span> <span class="va">alpha</span>,
  gamma <span class="op">=</span> <span class="va">gamma</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="posterior-sampling" class="section level2">
<h2 class="hasAnchor">
<a href="#posterior-sampling" class="anchor"></a>Posterior sampling</h2>
<p>We need to estimate <span class="math inline">\(\theta\)</span> and <span class="math inline">\(B = \left[\beta_{1}, \cdots, \beta_{K} \right]^{T}.\)</span></p>
<p>We estimate the parameters using HMC NUTS with four chains and 2000 iterations. Out of these 2000 iterations, 1000 iterations were used as warm-up samples.</p>
<p>We use high-performance computing and save the results.</p>
<p>We specify a file name.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fileN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span>
  <span class="st">"JABES_Endo_all_K_"</span>,
  <span class="va">K</span>,
  <span class="st">"_ite_"</span>,
  <span class="va">iter</span>,
  <span class="st">".RData"</span>
  <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">iter</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="va">chains</span> <span class="op">&lt;-</span> <span class="fl">4</span>

<span class="va">stan.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/LDAtopicmodel.html">LDAtopicmodel</a></span><span class="op">(</span>
  stan_data <span class="op">=</span> <span class="va">stan.data</span>, 
  iter <span class="op">=</span> <span class="va">iter</span>, 
  chains <span class="op">=</span> <span class="va">chains</span>, 
  sample_file <span class="op">=</span> <span class="cn">NULL</span>,
  diagnostic_file <span class="op">=</span> <span class="cn">NULL</span>,
  cores <span class="op">=</span> <span class="fl">4</span>,
  control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>adapt_delta <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>,
  save_dso <span class="op">=</span> <span class="cn">TRUE</span>,
  algorithm <span class="op">=</span> <span class="st">"NUTS"</span>
  <span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/save.html">save</a></span><span class="op">(</span><span class="va">stan.fit</span>, file <span class="op">=</span> <span class="va">fileN</span><span class="op">)</span></code></pre></div>
<p>We extract posterior samples from the results.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu">rstan</span><span class="fu">::</span><span class="fu"><a href="https://mc-stan.org/rstan/reference/stanfit-method-extract.html">extract</a></span><span class="op">(</span>
  <span class="va">stan.fit</span>, 
  permuted <span class="op">=</span> <span class="cn">TRUE</span>, 
  inc_warmup <span class="op">=</span> <span class="cn">FALSE</span>, 
  include <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span></code></pre></div>
<div id="alignment" class="section level3">
<h3 class="hasAnchor">
<a href="#alignment" class="anchor"></a>Alignment</h3>
<p>We estimated the parameters using the HMC-NUTS sampler with four chains and 2000 iterations. Out of these 2000 iterations, 1000 iterations were used as warm-up samples and discarded. Label switching across chains makes it difficult to directly compute log predictive density, split-<span class="math inline">\(\hat{R}\)</span>, effective sample size for model assessment, and evaluate convergence and mixing of chains.</p>
<p>To address this issue, we fixed the order of topics in chain one and then found the permutation that best aligned the topics across all four chains. For each chain two to four, we identified the estimated topics pair with the highest correlation, then found the next highest pair among the remaining, and so forth.</p>
<p>We create a Topic <span class="math inline">\(*\)</span> Chain matrix.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">theta</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">$</span><span class="va">theta</span> 

<span class="va">aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/alignmentMatrix.html">alignmentMatrix</a></span><span class="op">(</span>
  <span class="va">theta</span>, 
  <span class="va">ps</span>, 
  <span class="va">K</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>,
  chain <span class="op">=</span> <span class="va">chains</span>,
  SampleID_name <span class="op">=</span> <span class="st">"unique_names"</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="align-topic-proportion" class="section level3">
<h3 class="hasAnchor">
<a href="#align-topic-proportion" class="anchor"></a>Align topic proportion</h3>
<p>We align the topic proportion in each specimen across four chains.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">theta_aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/thetaAligned.html">thetaAligned</a></span><span class="op">(</span>
  <span class="va">theta</span>, 
  <span class="va">K</span>, 
  <span class="va">aligned</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>, 
  chain <span class="op">=</span> <span class="va">chains</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="align-asv-proportion" class="section level3">
<h3 class="hasAnchor">
<a href="#align-asv-proportion" class="anchor"></a>Align ASV proportion</h3>
<p>We align the ASVs proportion in each topic across four chains.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">beta</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">$</span><span class="va">beta</span> 


<span class="va">beta_aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/betaAligned.html">betaAligned</a></span><span class="op">(</span>
  <span class="va">beta</span>, 
  <span class="va">K</span>, 
  <span class="va">aligned</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>, 
  chain <span class="op">=</span> <span class="va">chains</span>
  <span class="op">)</span> </code></pre></div>
</div>
</div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h2>
<div id="plot-topic-proportion" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-topic-proportion" class="anchor"></a>Plot topic proportion</h3>
<p>Plot the topic proportion in specimens and we can save it for publication purposes. We can draw an informative summary of bacterial communities in each phenotype.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotTopicProportion</span>(</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  ps,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  theta_aligned,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  K,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">col_names_theta_all =</span> <span class="fu">c</span>(<span class="st">"iteration"</span>, <span class="st">"Sample"</span>, <span class="st">"Topic"</span>, <span class="st">"topic.dis"</span>),</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">chain =</span> <span class="dv">4</span>,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">iterUse =</span> iter<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">SampleIDVarInphyloseq =</span> <span class="st">"unique_names"</span>,</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> pna</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(<span class="st">"topic_dis_all_"</span>,K, <span class="st">".png"</span>)), p, width <span class="ot">=</span> <span class="dv">30</span>, height <span class="ot">=</span> <span class="dv">16</span><span class="er">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="topic_dis_all_11.png" alt="Figure 6: Topic distribution in specimens from Sonchus Arvensis plant. The color gradient represents the median topic distribution." width="70%"><p class="caption">
Figure 6: Topic distribution in specimens from Sonchus Arvensis plant. The color gradient represents the median topic distribution.
</p>
</div>
</div>
<div id="plot-asvs-distribution" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-asvs-distribution" class="anchor"></a>Plot ASVs distribution</h3>
<p>We plot the ASVs distribution in each topic.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/plotASVCircular.html">plotASVCircular</a></span><span class="op">(</span>
  <span class="va">ps</span>,
  <span class="va">K</span>,
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>,
  chain <span class="op">=</span> <span class="va">chains</span>,
  <span class="va">beta_aligned</span>,
  varnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"iterations"</span>, <span class="st">"topic"</span>, <span class="st">"rsv_ix"</span><span class="op">)</span>,
  value_name <span class="op">=</span> <span class="st">"beta_h"</span>,
  taxonomylevel <span class="op">=</span> <span class="st">"Class"</span>,
  thresholdASVprop <span class="op">=</span> <span class="fl">0.008</span>
  <span class="op">)</span>

<span class="fu">ggsave</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"circular_plot_plant_all_K_"</span>, <span class="va">K</span>, <span class="st">".png"</span><span class="op">)</span>, <span class="va">p</span>, width <span class="op">=</span> <span class="fl">28</span>, height <span class="op">=</span> <span class="fl">22</span><span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="circular_plot_plant_all_K_11.png" alt="Figure 7: ASV distribution over topics in all specimens" width="100%"><p class="caption">
Figure 7: ASV distribution over topics in all specimens
</p>
</div>
</div>
</div>
<div id="model-assessement" class="section level2">
<h2 class="hasAnchor">
<a href="#model-assessement" class="anchor"></a>Model assessement</h2>
<p>We do goodness of fit test using the posterior estimates and observed data.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_hist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/modelAssessment.html">modelAssessment</a></span><span class="op">(</span>
  <span class="va">ps</span>,
  <span class="va">stan.fit</span>,
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>,
  statistic <span class="op">=</span> <span class="st">"max"</span>,
  ASVsIndexToPlot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">10</span><span class="op">:</span><span class="fl">14</span>, <span class="fl">19</span><span class="op">:</span><span class="fl">26</span>, <span class="fl">36</span>, <span class="fl">51</span><span class="op">:</span><span class="fl">53</span>, <span class="fl">148</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu">ggsave</span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span>
    <span class="st">"model_assesment_hist_all_obs_sim_K_"</span>, 
    <span class="va">K</span>, 
    <span class="st">".png"</span>
    <span class="op">)</span>, 
  <span class="va">p_hist</span>, 
  width <span class="op">=</span> <span class="fl">12</span>, 
  height <span class="op">=</span> <span class="fl">6</span>
  <span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="model_assesment_hist_all_obs_sim_K_11.png" alt="Supplementary Figure 13: Predictive model check with simulated data, observed data, and a statistic G(Kij) = max{Kij}. Each facet shows the histogram of G(Kij) of each ASV in specimens from the posterior predictive distribution and the vertical line shows the value of G (Kij ) of each ASV in observed data." width="100%"><p class="caption">
Supplementary Figure 13: Predictive model check with simulated data, observed data, and a statistic G(Kij) = max{Kij}. Each facet shows the histogram of G(Kij) of each ASV in specimens from the posterior predictive distribution and the vertical line shows the value of G (Kij ) of each ASV in observed data.
</p>
</div>
</div>
<div id="diagnostics" class="section level2">
<h2 class="hasAnchor">
<a href="#diagnostics" class="anchor"></a>Diagnostics</h2>
<div id="effective-sample-size-ess" class="section level3">
<h3 class="hasAnchor">
<a href="#effective-sample-size-ess" class="anchor"></a>Effective sample size (ESS)</h3>
<p>We compute effective sample size for each parameter.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">diagnosticsPlot</span>(</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">theta_aligned =</span> theta_aligned,</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">beta_aligned =</span> beta_aligned,</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">iterUse =</span> iter<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">chain =</span> <span class="dv">4</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>p_ess_bulk <span class="ot">&lt;-</span> p[[[<span class="dv">1</span>]]]</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">"ess_bulk_plant_all_K_"</span>, K, <span class="st">".png"</span>), </span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  p_ess_bulk, </span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">9</span>, </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">6</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="ess_bulk_plant_all_K_11.png" alt="Supplementary Figure 14: Effective sample size (ESS) with eleven topics." width="100%"><p class="caption">
Supplementary Figure 14: Effective sample size (ESS) with eleven topics.
</p>
</div>
</div>
<div id="hatr" class="section level3">
<h3 class="hasAnchor">
<a href="#hatr" class="anchor"></a><span class="math inline">\(\hat{R}\)</span>
</h3>
<p>We can check whether <span class="math inline">\(\hat{R}\)</span> is less than 1.02.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_rhat</span> <span class="op">&lt;-</span> <span class="va">p</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>

<span class="fu">ggsave</span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"Rhat_plant_all_K_"</span>, <span class="va">K</span>, <span class="st">".png"</span><span class="op">)</span>, 
  <span class="va">p_rhat</span>, 
  width <span class="op">=</span> <span class="fl">9</span>, 
  height <span class="op">=</span> <span class="fl">6</span>
  <span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="Rhat_plant_all_K_11.png" alt="Supplementary Figure 15: Split Rhat with eleven topics." width="100%"><p class="caption">
Supplementary Figure 15: Split Rhat with eleven topics.
</p>
</div>
</div>
</div>
<div id="differential-topic-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-topic-analysis" class="anchor"></a>Differential topic analysis</h2>
<div id="test-on-all-specimens" class="section level3">
<h3 class="hasAnchor">
<a href="#test-on-all-specimens" class="anchor"></a>Test on all specimens</h3>
<p>We do test on all specimens. We can write DESeq2 results to LaTex.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/diffTopAnalysis.html">diffTopAnalysis</a></span><span class="op">(</span>
  design <span class="op">=</span> <span class="op">~</span> <span class="va">pna</span>,
  <span class="va">ps</span>,
  <span class="va">theta_aligned</span>,
  subsetSample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
  fitType <span class="op">=</span> <span class="st">"mean"</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span>
  <span class="va">dds_all</span>, 
  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"dds_all_K_"</span>, <span class="va">K</span>,<span class="st">".rds"</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"dds_all_K_11"</span><span class="op">)</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds_all_K_11</span><span class="op">)</span>
<span class="va">res</span> </code></pre></div>
<pre><code>## log2 fold change (MLE): pna O vs M 
## Wald test p-value: pna O vs M 
## DataFrame with 11 rows and 6 columns
##            baseMean log2FoldChange     lfcSE      stat      pvalue        padj
##           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
## Topic_1  12980.2689      -0.902957  0.954118 -0.946379 3.43955e-01 0.609485512
## Topic_2   1388.4912       0.328641  0.691155  0.475495 6.34434e-01 0.719579869
## Topic_3     14.5487      -0.559488  0.584374 -0.957414 3.38358e-01 0.609485512
## Topic_4   6138.5557       2.056851  0.749974  2.742564 6.09615e-03 0.022352555
## Topic_5     50.1848       2.693638  0.665150  4.049672 5.12894e-05 0.000564184
## Topic_6   5002.9019      -0.285674  0.795705 -0.359020 7.19580e-01 0.719579869
## Topic_7     25.1418      -0.382548  0.498956 -0.766696 4.43262e-01 0.609485512
## Topic_8    642.9452       0.638403  0.749673  0.851575 3.94450e-01 0.609485512
## Topic_9    728.3070       0.780513  0.781941  0.998174 3.18195e-01 0.609485512
## Topic_10  2452.9197       2.617431  0.725300  3.608757 3.07668e-04 0.001692175
## Topic_11  6788.9347       0.330308  0.782177  0.422294 6.72811e-01 0.719579869</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#writeResTable(res, fileN = paste0("dds_all_K_", K,".tex"))</span></code></pre></div>
</div>
<div id="test-on-paired-specimens" class="section level3">
<h3 class="hasAnchor">
<a href="#test-on-paired-specimens" class="anchor"></a>Test on paired-specimens</h3>
<p>We do the test on paired-specimens.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds_all_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/diffTopAnalysis.html">diffTopAnalysis</a></span><span class="op">(</span>
  design <span class="op">=</span> <span class="op">~</span> <span class="va">pna</span>,
  <span class="va">ps</span>,
  <span class="va">theta_aligned</span>,
  subsetSample <span class="op">=</span> <span class="va">paired_specimens</span>
<span class="op">)</span>
 
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span>
  <span class="va">dds_all_paired</span>, 
  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"dds_all_paired_K_"</span>, <span class="va">K</span>,<span class="st">".rds"</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"dds_all_paired_K_11"</span><span class="op">)</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds_all_paired_K_11</span><span class="op">)</span>
<span class="va">res</span> </code></pre></div>
<pre><code>## log2 fold change (MLE): pna O vs M 
## Wald test p-value: pna O vs M 
## DataFrame with 11 rows and 6 columns
##           baseMean log2FoldChange     lfcSE      stat      pvalue        padj
##          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
## Topic_1  3983.7697      -9.412600  1.056278 -8.911103 5.05275e-19 5.55802e-18
## Topic_2   898.6700       1.491941  0.999725  1.492352 1.35607e-01 2.13097e-01
## Topic_3    23.5763       0.282751  0.924598  0.305810 7.59749e-01 8.12226e-01
## Topic_4  1138.7952       3.371133  0.995722  3.385618 7.10182e-04 1.95300e-03
## Topic_5    31.7514       2.290040  0.892339  2.566333 1.02780e-02 1.88430e-02
## Topic_6    83.5973       0.351490  0.912658  0.385128 7.00143e-01 8.12226e-01
## Topic_7    35.3625       0.233201  0.801434  0.290979 7.71067e-01 8.12226e-01
## Topic_8   393.8606      -0.265216  1.116438 -0.237556 8.12226e-01 8.12226e-01
## Topic_9   468.2304      -6.004885  0.944671 -6.356589 2.06282e-10 7.56369e-10
## Topic_10 1938.0794       3.029427  1.085621  2.790502 5.26263e-03 1.15778e-02
## Topic_11  419.7516       6.324962  0.940052  6.728314 1.71641e-11 9.44024e-11</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#writeResTable(res, fileN = paste0("dds_all_paired_K_", K,".tex"))</span></code></pre></div>
</div>
<div id="test-on-paired-non-aster-specimens" class="section level3">
<h3 class="hasAnchor">
<a href="#test-on-paired-non-aster-specimens" class="anchor"></a>Test on paired non-Aster specimens</h3>
<p>We do the test on paired non-Aster specimens.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds_all_paired_nonaster</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/diffTopAnalysis.html">diffTopAnalysis</a></span><span class="op">(</span>
  design <span class="op">=</span> <span class="op">~</span> <span class="va">pna</span>,
  <span class="va">ps</span>,
  <span class="va">theta_aligned</span>,
  subsetSample <span class="op">=</span> <span class="va">paired_non_aster</span>
<span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">dds_all_paired_nonaster</span>, 
        file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"dds_all_paired_nonaster_K_"</span>, <span class="va">K</span>,<span class="st">".rds"</span><span class="op">)</span>
        <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"dds_all_paired_nonaster_K_11"</span><span class="op">)</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds_all_paired_nonaster_K_11</span><span class="op">)</span>
<span class="va">res</span> </code></pre></div>
<pre><code>## log2 fold change (MLE): pna O vs M 
## Wald test p-value: pna O vs M 
## DataFrame with 11 rows and 6 columns
##            baseMean log2FoldChange     lfcSE        stat      pvalue       padj
##           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;
## Topic_1   860.20984    -0.00301489  1.571767 -0.00191815 0.998469536 0.99846954
## Topic_2  1040.00766    -0.23883154  1.489550 -0.16033804 0.872614794 0.99846954
## Topic_3     4.72578    -1.96488007  1.042164 -1.88538517 0.059377854 0.16328910
## Topic_4   180.11478     4.32606625  1.204541  3.59146455 0.000328825 0.00361707
## Topic_5    27.56455     2.23657681  1.180879  1.89399299 0.058225941 0.16328910
## Topic_6     6.67586     0.09843585  0.881377  0.11168412 0.911073873 0.99846954
## Topic_7     5.19734    -0.59432231  0.892778 -0.66569966 0.505603089 0.79451914
## Topic_8   425.39352     0.63277929  1.509760  0.41912576 0.675124224 0.92829581
## Topic_9  2991.31585    -2.69366584  1.665720 -1.61711797 0.105852819 0.19406350
## Topic_10 1417.10667     2.41994794  1.495368  1.61829572 0.105598883 0.19406350
## Topic_11   33.61182     3.26394880  1.158670  2.81697937 0.004847764 0.02666270</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#writeResTable(res, fileN = paste0("dds_all_paired_nonaster_", K,".tex"))</span></code></pre></div>
</div>
<div id="test-on-paired-aster-specimens" class="section level3">
<h3 class="hasAnchor">
<a href="#test-on-paired-aster-specimens" class="anchor"></a>Test on paired Aster specimens</h3>
<p>We do the test on paired Aster specimens.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds_all_paired_aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/diffTopAnalysis.html">diffTopAnalysis</a></span><span class="op">(</span>
  design <span class="op">=</span> <span class="op">~</span> <span class="va">pna</span>,
  <span class="va">ps</span>,
  <span class="va">theta_aligned</span>,
  subsetSample <span class="op">=</span> <span class="va">paired_aster</span>
<span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span>
  <span class="va">dds_all_paired_aster</span>, 
  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"dds_all_paired_aster_K_"</span>, <span class="va">K</span>,<span class="st">".rds"</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"dds_all_paired_aster_K_11"</span><span class="op">)</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds_all_paired_aster_K_11</span><span class="op">)</span>
<span class="va">res</span> </code></pre></div>
<pre><code>## log2 fold change (MLE): pna O vs M 
## Wald test p-value: pna O vs M 
## DataFrame with 11 rows and 6 columns
##           baseMean log2FoldChange     lfcSE       stat      pvalue        padj
##          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
## Topic_1  6218.5368    -10.4304467   1.51290 -6.8943421 5.41147e-12 5.95261e-11
## Topic_2   808.8041      1.2913524   1.35518  0.9529025 3.40639e-01 6.24506e-01
## Topic_3    62.3447      0.5769415   1.41106  0.4088702 6.82635e-01 8.34332e-01
## Topic_4  3197.9976      0.9723636   1.53691  0.6326749 5.26946e-01 8.28058e-01
## Topic_5    10.5337     -0.1502755   1.27513 -0.1178511 9.06186e-01 9.48164e-01
## Topic_6   188.5282      0.0862107   1.32605  0.0650131 9.48164e-01 9.48164e-01
## Topic_7    20.2758      0.4522516   1.03245  0.4380355 6.61361e-01 8.34332e-01
## Topic_8    12.0123     -1.3189221   1.02924 -1.2814486 2.00036e-01 4.52427e-01
## Topic_9  1035.0590     -7.8134758   1.32604 -5.8923438 3.80756e-09 1.39610e-08
## Topic_10  129.1210     -1.5272929   1.20675 -1.2656214 2.05649e-01 4.52427e-01
## Topic_11 2344.2823      8.3837719   1.38843  6.0383030 1.55743e-09 8.56589e-09</code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#writeResTable(res, fileN = paste0("dds_all_paired_aster_", K,".tex"))</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pratheepa Jeganathan, Susan Holmes.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
